api/v1alpha1/amaltheasession_types.go

```diff
@@ -0,0 +1,265 @@
+	// hibernated by Amalthea if it is failing.
+	FailedSeconds int `json:"failedSeconds,omitempty"`
+	// +kubebuilder:default:=0
+	// +kubebuilder:validation:Minimum:=0
+	// Number of seconds where a server can be in hibernated state before
+	// it gets completely deleted. A value of zero indicates that hibernated servers
+	// will not be automatically be deleted by Amalthea after a period of time.
+	HibernatedSeconds int `json:"hibernatedSeconds,omitempty"`
+}
+
+// +kubebuilder:validation:Enum={token,oauth2proxy}
+type AuthenticationType string
+
+const Token AuthenticationType = "token"
+const Oidc AuthenticationType = "oauth2proxy"
+
+type Authentication struct {
+	Enabled bool               `json:"enabled,omitempty"`
+	Type    AuthenticationType `json:"type,omitempty"`
+	// Kubernetes secret that contains the authentication configuration
+	// For `token` generate a hard to guess string / password-like string.
+	// this value can be used as Authorization header or as a cookie with the name `amaltheaSessionToken` to
+	// access the session.
+	// For `oauth2proxy` please see https://oauth2-proxy.github.io/oauth2-proxy/configuration/overview#config-file.
+	SecretRef SecretRef `json:"secretRef,omitempty"`
+}
+
+// A reference to a Kubernetes secret and a specific field in the secret to be used in a session
+type SecretRef struct {
+	Name string `json:"name"`
+	Key  string `json:"key"`
+}
+
+// +kubebuilder:validation:Enum={Running,Failed,Hibernated,NotReady,RunningDegraded}
+type State string
+
+const Running State = "Running"
+const Failed State = "Failed"
+const Hibernated State = "Hibernated"
+const NotReady State = "NotReady"
+const RunningDegraded State = "RunningDegraded"
+
+type ContainerCounts struct {
+	Ready int `json:"ready,omitempty"`
+	Total int `json:"total,omitempty"`
+}
+
+// AmaltheaSessionStatus defines the observed state of AmaltheaSession
+type AmaltheaSessionStatus struct {
+	// Conditions store the status conditions of the AmaltheaSessions. This is a standard thing that
+	// many operators implement see https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties
+	// +operator-sdk:csv:customresourcedefinitions:type=status
+	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type" protobuf:"bytes,1,rep,name=conditions"`
+	// +kubebuilder:default:=NotReady
+	State               State           `json:"state,omitempty"`
+	URL                 string          `json:"url,omitempty"`
+	ContainerCounts     ContainerCounts `json:"readyCounts,omitempty"`
+	InitContainerCounts ContainerCounts `json:"initReadyCounts,omitempty"`
+	Idle                bool            `json:"idle,omitempty"`
+	IdleSince           metav1.Time     `json:"idleSince,omitempty"`
+	FailingSince        metav1.Time     `json:"failingSince,omitempty"`
+	HibernatedSince     metav1.Time     `json:"hibernatedSince,omitempty"`
```

Comment:
I was wondering whether we should also have an `Anonymous` field ?

Reply:
@sgaist what would this indicate? That the user using the session is anonymous? This can happen in 2 cases potentially - either when you dont have authentication or you have the token based authentication. And token based authentication does not guarantee that the user is anonymous. So due to this ambiguity I am not sure that we should judge this here. Someone can look at the CR and make this determination based on what their use case is.